packages/datadog-instrumentations/src/apollo-gateway.js

```diff
@@ -0,0 +1,110 @@
+const {
+  addHook,
+  channel
+} = require('./helpers/instrument')
+const shimmer = require('../../datadog-shimmer')
+const tracingChannel = require('dc-polyfill').tracingChannel
+
+const validateCh = tracingChannel('apm:apollo-gateway:validate')
+const requestCh = tracingChannel('apm:apollo-gateway:request')
+const fetchCh = tracingChannel('apm:apollo-gateway:fetch')
+const planCh = tracingChannel('apm:apollo-gateway:plan')
+const executeCh = tracingChannel('apm:apollo-gateway:execute')
+const postProcessingCh = tracingChannel('apm:apollo-gateway:postprocessing')
+
+const generalErrorCh = channel('apm:apollo-gateway:general:error')
+const REQUEST_CTX = {}
```

Comment:
Any way this could be removed? Generally speaking globals should only be used when there is no other way to achieve the same behaviour.

Reply:
we can try to move the constant to the plugin side? have it publish the data before span is started store it on the plugin side and then add it to the span on startSpan for the plugin